### PR TITLE
Make 'measuring installs' a repeatable, modular piece of code (Fixes #6871)

### LINKED ISSUE #6871: Make "measuring installs" a repeatable, modular piece of code
(aka stub-attribution sidestep)

## Description

We're going to be measuring installs for some number of experiments in 2019, which requires us to change how we handle stub attribution code. [We've done that in a bespoke way](https://github.com/mozilla/bedrock/issues/6704#issuecomment-459357095). This makes it repeatable.

### Estimates 
(How much time in hours (up to 2), then in 1/2 day increments it will take each role to complete the work required to meet the Success Criteria established for this issue).

| Estimates |  |
| ------------- | ------------- |
| TL = TeamLead | |
| PM = Program Manager | |
| Eng = Engineer | 2 days |
| SEO | |
| ANA = Analytics | |
| DES = Design | |
| COPY | |

---

#### :yellow_heart: Success Criteria :yellow_heart:
- [ ] We can measure install rate for people clicking on a particular download button
- [ ] We do not break any upstream utm parameters
    * If there is no utm_content param, then we can add it, with a value that includes the experiment identifier and the variation identifier as in `utm_content=download_thanks_install_experiment-v-{a|b|c|d}`
    * If we add a utm_content param, we must ensure that utm_source, utm_medium and utm_content are also populated. If any are missing then we can fill them with default values, e.g. `utm_source=www.mozilla.org`, `utm_medium=download_button{or other ui treatment}`, `utm_campaign=download_thanks_page{or other campaign}`.
    * We should not overwrite utm parameters that exist. 

#### :heavy_exclamation_mark: Risks :heavy_exclamation_mark:
- [ ] This is an example of a risk. Use a checklist!
- [ ] Product may not include feature X by release day Y. . .

#### Tasks
- [ ] This is an example of a task. Use a checklist!
- [ ] Code the page using these assets supplied by team x

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Adds a script that enables passing custom data to the Stub Attribution service for first-party experiments.
- Adds documentation on how Stub Attribution works.
- Converts the "how to install" experiment code to use the new script (this experiment is finished in prod, but I figured this would be a good way to test locally that this all works).

## Issue / Bugzilla link
#6871

## Testing
- [x] "How to install" experiment still functions as expected using the new code.
  - Open `/firefox/new/` using Chrome on Windows.
  - Once redirected to a variation, inspect cookies in dev tools and copy the value for `moz-stub-attribution-code`.
  - [Decode](https://www.base64decode.org/) that base64 value, and it should match the params configured in the experiment (e.g. `source=www.mozilla.org&medium=download_button&campaign=download_thanks_page&content=download_thanks_install_experiment-v-a`).

### CODE REVIEW

File: docs/stub-attribution.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+.. This Source Code Form is subject to the terms of the Mozilla Public
+.. License, v. 2.0. If a copy of the MPL was not distributed with this
+.. file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+.. _stub_attribution:
+
+================
+Stub Attribution
```
Comment (stephaniehobson) [inline docs/stub-attribution.rst]: DOCS 😍

**Comment by stephaniehobson** [comment]
I'm getting a 403 from `/firefox/stub_attribution_code/` when I try to test on demo. Is this expected to work on a demo server?

File: docs/stub-attribution.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+the Windows stub installer, where it finally gets passed to Firefox for use in
+Telemetry. This attribution funnel enables Mozilla to better understand how
+different marketing campaigns effect overall retention in Firefox.
+
+How does it work in bedrock?
+----------------------------
+
+The base template in bedrock contains a stub attribution script that runs on every
+page of the website. The script only runs on Windows (since the stub installer is a
+Windows feature) and if cookies are enabled. The script also respects Do Not Track,
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply docs/stub-attribution.rst]: I'm wrong. It's the funnelcakes that work for all systems not the stub installer.

**Comment by alexgibson** [comment]
> I'm getting a 403 from /firefox/stub_attribution_code/ when I try to test on demo. Is this expected to work on a demo server?

Ah, I always forget there's an environment variable that needs setting and this doesn't work by default. Let me add it to the docs 👍

**Comment by alexgibson** [comment]
@stephaniehobson pushed another commit with some updates to the docs.

**Pull request merged**